packages/@aws-cdk/aws-rds/test/cluster.test.ts

```diff
@@ -760,7 +760,55 @@ describe('cluster', () => {
     cluster.addRotationSingleUser();
 
     // THEN
-    expect(() => cluster.addRotationSingleUser()).toThrow(/A single user rotation was already added to this cluster/);
+    expect(() => cluster.addRotationSingleUser()).toThrow(/A single user rotation for master secret is already added to this cluster/);
+
+
+  });
+
+  test('throws when trying to add single user rotation for additional secret which is master secret', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.AURORA_MYSQL,
+      credentials: { username: 'admin' },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+    });
+
+    // WHEN
+    cluster.addRotationSingleUser();
+
+    // THEN
+    expect(() => cluster.addRotationSingleUser({ secret: cluster.secret!! })).toThrow(/A single user rotation for master secret is already added to this cluster/);
+
+
+  });
+
+  test('throws when trying to add single user rotation for additional secret multiple times', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.AURORA_MYSQL,
+      credentials: { username: 'admin' },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+    });
+    const secret = new DatabaseSecret(stack, 'AdditionalSecret', {
+      username: 'myuser',
+    });
+    secret.attach(cluster);
+
+    // WHEN
+    cluster.addRotationSingleUser({ secret });
+
+    // THEN
+    expect(() => cluster.addRotationSingleUser({ secret })).toThrow(/A single user rotation for additional secret is already added to this cluster/);
 
 
   });
```

Comment:
I feel like we're missing a positive-case test.

Reply:
I couldn't find any positive-case test for current behavior and thought it should be covered by other tests or something.

Reply:
Can you point me one to learn how to write it.

Reply:
Sure. Here's an example: https://github.com/aws/aws-cdk/blob/7176a5d5208da7d727bbf5112bc12533983380ea/packages/%40aws-cdk/aws-rds/test/instance.test.ts#L167-L195

Reply:
This is the secret itself not the rotation. The secret may or may not be created in the same stack. I can write one but it's gonna be different from this positive test, since it requires additional resources.

Reply:
Thanks for the link. Will do it. Just to let you know, I'm blocked by [this](https://github.com/aws/aws-cdk/pull/13799#discussion_r604007605).